src/ECS/ECS.cpp

```diff
@@ -0,0 +1,179 @@
+            Rendering::Render(*assetFile->GetTextureAtlas(), px, py, &r);
+        }
+
+        void render (entt::registry &registry) {
+            auto view = registry.view<const EntityTypeComponent, const PositionComponent, const AnimatedComponent>();
+
+            // this is just getting pointers to preloaded, parsed file data
+            // but maybe it can be moved somewhere else if this is performance hit
+            auto lordFile = AssetStorage::getFileGM1(AssetStorageKey::BODY_LORD_GM1);
+            auto deerFile = AssetStorage::getFileGM1(AssetStorageKey::BODY_DEER_GM1);
+
+            auto appleTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_APPLE_GM1);
+            auto birchTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_BIRCH_GM1);
+            auto chestnutTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_CHESTNUT_GM1);
+            auto oakTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_OAK_GM1);
+            auto pineTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_PINE_GM1);
+            auto shrub1TreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_SHRUB1_GM1);
+            auto shrub2TreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_SHRUB2_GM1);
+
+            for(auto [entity, typeComponent, positionComponent, animationComponent]: view.each()) {
+                switch (typeComponent.type) {
+                    case EntityType::LORD: renderUnit(positionComponent, animationComponent, lordFile); break;
+                    case EntityType::DEER: renderUnit(positionComponent, animationComponent, deerFile); break;
+
+                    case EntityType::TREE_APPLE: renderUnit(positionComponent, animationComponent, appleTreeFile); break;
+                    case EntityType::TREE_BIRCH: renderUnit(positionComponent, animationComponent, birchTreeFile); break;
+                    case EntityType::TREE_CHESTNUT: renderUnit(positionComponent, animationComponent, chestnutTreeFile); break;
+                    case EntityType::TREE_OAK: renderUnit(positionComponent, animationComponent, oakTreeFile); break;
+                    case EntityType::TREE_PINE: renderUnit(positionComponent, animationComponent, pineTreeFile); break;
+                    case EntityType::TREE_SHRUB1: renderUnit(positionComponent, animationComponent, shrub1TreeFile); break;
+                    case EntityType::TREE_SHRUB2: renderUnit(positionComponent, animationComponent, shrub2TreeFile); break;
+                }
+            }
+        }
+
+        void updateAnimationFrameIndexes (entt::registry &registry, double now) {
+            auto view = registry.view<const EntityTypeComponent, AnimatedComponent>();
+            
+            for(auto [entity, typeComponent, animationComponent]: view.each()) {
+                animationComponent.frameIndex = (24 - abs(int(now * 15.0f) % (2 * 24) - 24));
```

Comment:
If you put raw numbers, please put them in a constexpr variable. 
Nobody will understand what they mean.

Reply:
tbh - EnTT docs are recommending doing that in this way, but I will switch that in further work

Reply:
tbh - EnTT docs are recommending doing that in this way, but I will switch that in further work